userspace/libsinsp/test/filter_compiler.ut.cpp

```diff
@@ -681,3 +756,29 @@ TEST_F(sinsp_with_test_input, filter_transformers_wrong_input_type)
 	ASSERT_THROW(evaluate_filter_str(&m_inspector, "b64(evt.rawres) = -1", generate_getcwd_failed_entry_event()),
 		     sinsp_exception);
 }
+
+TEST_F(sinsp_with_test_input, filter_regex_operator_evaluation)
+{
+	// Basic case just to assert that the basic setup works
+	add_default_init_thread();
+	open_inspector();
+
+	auto evt = generate_getcwd_failed_entry_event();
+
+	// legit use case with a string
+	EXPECT_TRUE(evaluate_filter_str(&m_inspector, "evt.source regex '^[s]{1}ysca[l]{2}$'", evt));
```

Comment:
Note that in https://github.com/falcosecurity/libs/pull/1911 I'm refactoring these kind of tests to be like:

```cpp
EXPECT_TRUE(eval_filter(evt, "evt.source regex '^[s]{1}ysca[l]{2}$'"));
```

The change to these will be very quick to do.

Reply:
Can we make the changes since #1911 was merged?
This LGTM then :)

Reply:
I apologize for having lost these comments entirely! Let me rebase and push the changes as suggested.